### ISSUE TITLE
Add docs for different config profiles

### ISSUE THREAD

**Issue by drpatelh** [nf-core/configs#6]
Custom configs are now hosted centrally at `nf-core/configs`. We need to think about where we host the documentation for running these config files. These are currently added on a per-pipeline basis but it would be good to also host them centrally so they are available to all pipelines.

This will also probably involve refactoring the current docs to deal with this.

Documention to-do list:

- [x] BINAC (Alex P)
- [x] CCGA 
- [x] CFC (Alex P)
- [x] GIS
- [x] hebbe
- [x] phoenix (Alex P)
- [x] uct_hex
- [x] uppmax_devel
- [x] uppmax
- [x] uzh (Alex P)

**Comment by ewels** [comment]
Note that I think `nf-core/configs/docs/` would be fine for organisation. Could have a `TEMPLATE.md` if we really want? Or people can just look at the existing files.

**Comment by drpatelh** [comment]
I have added the docs for the Crick in the `nf-core/configs/docs/` folder. See https://github.com/nf-core/configs/pull/7 

We need to add docs for the rest of the configs currently hosted in the repository for consistency. 

Also, need to reference the documentation for these profiles in `nf-core/tools` template. This is what we have so far:
https://github.com/nf-core/tools/blob/dev/nf_core/pipeline-template/%7B%7Bcookiecutter.name_noslash%7D%7D/docs/usage.md#custom-resource-requests

Im not sure whether its possible to get a dynamic listing of all the available profiles available to use when running the pipeline but that would be quite neat.

**Comment by apeltzer** [comment]
Neat! Something like a git submodule that we integrate into the `nf-core/tools` package maybe? Not been using these too often, but it might be possible.

**Comment by apeltzer** [comment]
- [ ] BINAC (Alex P)
- [ ] CCGA 
- [ ] CFC (Alex P)
- [ ] GIS
- [ ] hebbe
- [ ] phoenix (Alex P)
- [ ] uct_hex
- [ ] uppmax_devel
- [ ] uppmax
- [ ] uzh (Alex P)

**Comment by drpatelh** [comment]
Thanks @apeltzer!

I was thinking it would be nice if we could maybe automatically add the list of available profiles to the help string produced by the pipeline. Maybe we can add an extra description document in `nf-core/configs` that contains the profile name and a brief help comment. This can then be parsed in nextflow.config and rendered in the help.

I'm not sure if nextflow currently checks that the name of the profile has to be valid. I'll test that.

**Comment by matq007** [comment]
It would be super cool to maybe specify a little more how to use this config files in an existing/new nfcore solution. @apeltzer has a really nice code in [nextflow.config](https://github.com/nf-core/eager/blob/962eb96e55dabc745727929497adf4e4e4a27317/nextflow.config).

**Comment by ewels** [comment]
Deleted your comment @apeltzer and added the list to the original post by @drpatelh so that we get a nice completion tick-list on the issues page.

**Comment by mihai-sysbio** [comment]
For `Uppmax` #53 and `Hebbe` #54 the CI seems to report the same issue. Is there a mismatch between the documentation and the changes in the PRs?

**Comment by abhi18av** [comment]
Considering that the list in the first comment is now completed already I am adding a list of configs (from https://github.com/nf-core/configs/tree/master/conf)  which need to be documented still 


- [] phoenix.config
- [] cfc_dev.config
- [] gis.config
- [] hebbe.config


Here's the [template.md](https://github.com/nf-core/configs/blob/master/docs/template.md) which can be used to create the docs for configs above.

**Comment by maxulysse** [comment]
Friendly reminder:

@FriederikeHanssen we're missing docs for the `cfc_dev` config
@athbaltzis we're missing docs for the `crg` config
@ewels we're missing docs for the `hebbe` config
@yassineS and @apeltzer we're missing docs for the `phoenix` config

**Comment by apeltzer** [comment]
Yeah thats one for @yassineS - can't work on this at the moment unforrtunately as I'm not even near that cluster ;-)

**Comment by maxulysse** [comment]
> Yeah thats one for @yassineS - can't work on this at the moment unforrtunately as I'm not even near that cluster ;-)

You're still listed as contact in the config, that's how I found you

**Comment by yassineS** [comment]
The cluster has been replaced by a new one, hpc1, but I will let @roberta-davidson comment. She is our resident expert now on all eager matters.

**Comment by jfy133** [comment]
Let me know if you need any advice/help @roberta-davidson!

**Comment by mihai-sysbio** [comment]
> Considering that the list in the first comment is now completed already I am adding a list of configs (from https://github.com/nf-core/configs/tree/master/conf) which need to be documented still

The list should be edited to replace `hebbe` by `vera` ([docs](https://www.c3se.chalmers.se/)).

**Comment by maxulysse** [comment]
I think this is done and can be closed now, also we have a website now: https://nf-co.re/configs/
